### PR TITLE
Revert "graphviz: update to 2.42.2."

### PR DESCRIPTION
**Pull request by ndowens**
Reverting to older version, until I can solve a discovered link issue that was mentioned and I was able to reproduce

This reverts commit f0fb795c17982f6a93aa6c27a86e349b825e9b02.

### CODE REVIEW

**Comment by ndowens** [comment]
I've been working on getting the new version to work.
We have two options, I can remove the problem, is that it generates header during build, so:
1. No cross
2. We use the header from this version into the newer version.

**Comment by ndowens** [comment]
2nd method does allow cross-builds. https://termbin.com/mf5j  is the header that is suppose to be generated during build

File: srcpkgs/graphviz/patches/cross.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--- lib/gvpr/Makefile.am.orig
++++ lib/gvpr/Makefile.am
+@@ -52,8 +52,7 @@ libgvpr_la_DEPENDENCIES = $(libgvpr_C_la
+ 
+ mkdefs_SOURCES = mkdefs.c
+ 
+-gdefs.h: $(top_srcdir)/lib/gvpr/gprdata mkdefs$(EXEEXT)
+-	./mkdefs$(EXEEXT) gdefs.h < $(top_srcdir)/lib/gvpr/gprdata
```
Comment (pullmoll) [inline srcpkgs/graphviz/patches/cross.patch]: It should be possible to add `build_helper="qemu"` to the template and patch this line to
```
        qemu-${XBPS_TARGET_QEMU_MACHINE}-static ./mkdefs$(EXEEXT) gdefs.h < $(top_srcdir)/lib/gvpr/gprdata
```
in e.g. `pre_configure()` and inside a block
```
        if [ "$CROSS_BUILD" ]; then
                vsed -i lib/gvpr/Makefile.am -e 's;\./mkdefs;qemu-${XBPS_TARGET_QEMU_MACHINE}-static &;'
        fi
```
Then cross builds will run `mkdefs` under `qemu`.

**Comment by jnbr** [comment]
switch to cmake was reverted, graphviz was update to 2.44.0.

**Pull request closed without merging**